### DISCUSSION TITLE
SelectPickers loading paginated data when scrolled to bottom.

### DISCUSSION THREAD

**Discussion by abdullaev388** [rsuite/rsuite#3088] [Q&A] [answered]
I want to implement infinite loading with SelectPicker. But i faced some issues. Rsuite uses react-window for virtualization. This library is good for fixed sizes, but when there are dynamic size items, it does not provide solution. As a result, i have a problem when my items have different height. You can open this example to see the problem: https://codesandbox.io/s/twilight-pond-5rbgur?file=/src/App.js

Well i can suggest moving from react-window to a @tanstack/virtual: https://tanstack.com/virtual/v3
This is a modern library for virtualization and most important it supports dynamic sizes. I want you to have a look on this library.

But also i assume another solution. Avoid using virtualization, and just somehow handle list scrolling, and fetch items when scrolled to bottom. So my question is, is there any way to do it?

**Comment by SevenOutman** [comment, marked as answer]
Hi @abdullaev388 , as for the virtualization issue, I would suggest use `virtualized={false}` to disable virtualization if your data items were not as many as thousands.

As for the scroll-reach-bottom callback, we don't have a callback for that at present. I think it's achievable with IntersectionObserver API. Considering the browser compatibility of IntersectionObserver, I can't promise we'd add such callback in v5, but it's hopefully to be added in v6 where we're going to drop IE support.

**Reply by abdullaev388** [reply]
Ok, but i need to implement infinite-loading somehow, can you tell me are there any ways to do it without using virtualization?

**Reply by SevenOutman** [reply]
I don't have done it in real projects but I think simply appending a "Load more" button to the end of the list (using `renderMenu` or something) should work. You don't have to stick on triggering fetching automatically.
